### ISSUE TITLE
Add the option to have locale prefix

### ISSUE THREAD

**Issue by Chenalejandro** [stack-auth/stack#320]
Currently, we can give the StackProvider a `lang` prop to set the language. However,  there is no i18n routing when navigating, for example, from the sign-in page to the sign-up page.

![Screenshot from 2024-10-24 19-26-23](https://github.com/user-attachments/assets/3e6432e2-4f5b-4911-b31a-217f66ed9f46)
![Screenshot from 2024-10-24 19-26-07](https://github.com/user-attachments/assets/af0bb543-7048-4e36-8122-1b995bb7cfe2)

I'm using next-intl with i18n routing. It would be nice to pass the locale (or whatever it is called) to the `StackProvider`, so it can add that as a prefix in all routes from the stack handler.

![image](https://github.com/user-attachments/assets/43c7427b-9cda-4c1b-9cf5-ffa528eecc91)

**Comment by madster456** [comment]
Closing as duplicate of the i18n overhaul meta issue #1047 .

The current localization system is being completely replaced, which will resolve the vast majority of translation bugs and enable proper support for new languages.

Once the rewrite is complete, any remaining problems will be re-opened or filed as fresh issues.

→ See #1047  for status and progress.

Thank you for the report! 🌍
